### ISSUE TITLE
Graphics corruption in latest build

### ISSUE THREAD

**Issue by dericksnow** [Ryujinx/Ryujinx#5685]
### Game Name

Red Dead Redemption

### Game Update Version

1.0.1

### Emulator Build Version

1.1.1014

### Graphics Backend

Vulkan

### Hardware Specs

 Model Name:	MacBook Pro
  Model Identifier:	MacBookPro18,3
  Model Number:	MKGP3LL/A
  Chip:	Apple M1 Pro
  Total Number of Cores:	8 (6 performance and 2 efficiency)
  Memory:	16 GB
  System Firmware Version:	8422.141.2
  OS Loader Version:	8422.141.2

### Game State

New graphics corruption compared to the previous build on Mac 1.1.1010

### Screenshots

Previous Build renders correctly on Mac
![1 1 1010-correct](https://github.com/Ryujinx/Ryujinx-Games-List/assets/60944464/e1e4ced4-48b1-42f5-b44b-8388e2495214)

current build on my mac with same default settings
![1 1 1014-incorrect](https://github.com/Ryujinx/Ryujinx-Games-List/assets/60944464/19db6836-b1f9-40a0-aec2-deb06bded0c6)

### Log File

[Ryujinx_1.1.1014_2023-09-14_17-02-14.log](https://github.com/Ryujinx/Ryujinx-Games-List/files/12613918/Ryujinx_1.1.1014_2023-09-14_17-02-14.log)

**Comment by viiibe** [comment]
Can reproduce on my M1 Air 8-core.

**Comment by shinra-electric** [comment]
I can confirm too. Seems caused by #5603 

1.1.1013: 
<img width="1392" alt="Screenshot 2023-09-15 at 17 14 39" src="https://github.com/Ryujinx/Ryujinx/assets/50119606/b5988f3b-1b0d-4fdd-b273-6cd23fd2e7d7">

1.1.1014 (Look at the hair): 
<img width="1392" alt="Screenshot 2023-09-15 at 17 16 38" src="https://github.com/Ryujinx/Ryujinx/assets/50119606/bac11c20-14f6-4fed-bce0-5ff471938f92">

**Comment by appvitor** [comment]
I can assure that this problem persists on build 1.1.1015, Red Dead Redemption has horrible glitches after this update.

[Ryujinx_1.1.1015_2023-09-16_16-19-06.log](https://github.com/Ryujinx/Ryujinx/files/12644441/Ryujinx_1.1.1015_2023-09-16_16-19-06.log)
[Ryujinx_1.1.1015_2023-09-17_17-06-17.log](https://github.com/Ryujinx/Ryujinx/files/12644442/Ryujinx_1.1.1015_2023-09-17_17-06-17.log)
[Ryujinx_1.1.1015_2023-09-17_23-54-52.log](https://github.com/Ryujinx/Ryujinx/files/12644443/Ryujinx_1.1.1015_2023-09-17_23-54-52.log)

**Comment by jacksonff** [comment]
Hi, not sure if it's related but I'm getting some strange rendering in Pokemon Sc/Vl:

![ryujinx_capture_2023-09-20_23-30-40](https://github.com/Ryujinx/Ryujinx/assets/68671644/664d5ef2-1655-4468-b8a5-33bafb5206a6)

Tried from version 1.1.1012 to latest 1.1.1020, even the artifacts from the IsaacMarovitz' PR. Same results after purging shader cache and rebuilding PTC every time. The weirdest part is that I don't get any of these issues in Paldea. Only in the Kitakami area from the DLC. Last thing I tried was checking if my dumps were fine (they are) and updating system version from 16.0.3 to 16.1.0 but no luck either.

Running on Windows 10.
CPU: AMD FX-8350
GPU: RX 550 (GCN 4.0 with AMD 23.9.1 drivers).
16GB RAM.
Vulkan backend.
